olive/evaluator/olive_evaluator.py

```diff
@@ -28,12 +29,12 @@ def get_metric(self, metric_name: str) -> Metric:
                 return metric
         raise ValueError(f"Metric {metric_name} not found")
 
-    def evaluate(self, model: OliveModel, target: OliveSystem = LocalSystem()) -> Dict:
+    def evaluate(self, model: OliveModel, target: OliveSystem = LocalSystem()) -> MetricResult:
         return target.evaluate_model(model, self.metrics)
 
 
 class OliveEvaluatorConfig(ConfigBase):
-    metrics: List[Metric]
+    metrics: MetricList
 
     @validator("metrics")
     def validate_metrics(cls, v):
```

Comment:
I just realized that since we are only validating the metrics in this config class, the check is not done if the evaluator is created directed using the `OliveEvaluator` pass. The evaluation tests directly create the evaluator object so they won't have this validation too. 

maybe we can abstract the `validate_metric` function and use it both here and in `OliveEvaluator` init?

Reply:
Yes, actually not only for Evaluator. The same problems for Metric, RunConfig. Will sent the seprated PR to fix that.

Reply:
https://github.com/microsoft/Olive/pull/256 Looks like it was fixed in this PR already.